app/src/main/java/lt/vilnius/tvarkau/NewProblemActivity.java

```diff
@@ -217,7 +219,15 @@ public void sendProblem() {
                         try {
                             bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), uri);
                             ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-                            bitmap.compress(Bitmap.CompressFormat.JPEG, 60, byteArrayOutputStream);
+                            Bitmap scaledBitmap;
+                            if (bitmap.getHeight() > bitmap.getWidth()){
+                                scaledBitmap = BitmapUtils.scaleToFitHeight(bitmap, 1000);
+                                bitmap.recycle();
+                            } else {
+                                scaledBitmap = BitmapUtils.scaleToFitWidth(bitmap, 1000);
+                                bitmap.recycle();
+                            }
+                            scaledBitmap.compress(Bitmap.CompressFormat.JPEG, 50, byteArrayOutputStream);
```

Comment:
I think you should move this code in `BitmapUtils` class and have already processed bitmap here.